multicluster/apis/multicluster/v1alpha1/clusterclaim_types.go

```diff
@@ -27,6 +27,8 @@ const (
 	WellKnownClusterClaimClusterSet = "clusterSet.k8s.io"
 )
 
+// +genclient
+// +genclient:nonNamespaced
```

Comment:
do we need new lines between genclient comments and kubebuilder tag comments? i remember there was a bug earlier in codegen which needed that line break but not sure if it has been resolved since. Have you tried to run the codegen to verify the listers/informers etc are correctly generated?

Reply:
no need to add new lines, I have tried the codes in a linux VM, it works fine. do you think I need to submit the generated code or leave it for now. I am thinking we may still need to change the CRD, so didn't checkin the auto-generated codes.

Reply:
Hi @abhiraut , I just updated the code and removed the line `// +genclient:nonNamespaced` for dedicated leader cluster use case.

Reply:
i think eventually we need the generated code for controller work, and it would be okay to update it if required since it is not merged in main

Reply:
@abhiraut , I just added those generated codes to this PR.